tmexp/create_bow.py

```diff
@@ -178,31 +170,29 @@ def create_bow(
             min_word_blob,
             max_word_blob,
         )
-        blacklisted_words = {
-            word
-            for word, count in doc_freq.items()
-            if count < min_word_blob or count > max_word_blob
-        }
+        blacklisted_words: WordCount = Counter()
+        for word, count in doc_freq.items():
+            if count < min_word_blob or count > max_word_blob:
+                blacklisted_words[word] = np.inf
         logger.info("Found %d words." % len(blacklisted_words))
-        logger.info("Pruning BOW...")
-        for repo, repo_bow in bow.items():
-            for doc_name, counts_list in repo_bow.items():
-                repo_bow[doc_name] = [
-                    {
-                        word: count
-                        for word, count in word_counts.items()
-                        if word not in blacklisted_words
-                    }
-                    for word_counts in counts_list
-                ]
-                docs[repo][doc_name] = [
-                    ref_list
-                    for i, ref_list in enumerate(docs[repo][doc_name])
-                    if repo_bow[doc_name][i]
-                ]
-                repo_bow[doc_name] = [
-                    word_counts for word_counts in repo_bow[doc_name] if word_counts
-                ]
+        logger.info("Pruning bags of words ...")
+        for doc_name in sorted(evolution_model):
+            doc_evolution = evolution_model.pop(doc_name)
+            pruned_evolution = DocumentEvolution(bows=[], refs=[])
+            for bow, refs in zip(doc_evolution.bows, doc_evolution.refs):
+                pruned_bow = bow - blacklisted_words
```

Comment:
It might be easier here to just use a `frozenset` for `blacklisted_words` and del the elements:
```python
for word in blacklisted_words:
    if word in bow:
        del bow[word]
```
It's a tad more verbose but more readable: even though I worked many times with `Counter` I had to check the effect of having `np.inf` as counts etc.

Reply:
done :)